cmd/clusternetwork/config.go

```diff
@@ -0,0 +1,70 @@
+package clusternetwork
+
+import (
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	operatorv1 "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterNetworkConfig interface contains methods to validate network configuration of a cluster
+type ClusterNetworkConfig interface {
+	Validate() error
+}
+
+// OVNKubernetes contains information specific to network type OVNKubernetes
+type OVNKubernetes struct {
```

Comment:
This can be a generic struct which can be specialized for OVN but I am okay leaving it for now as we are still trying to figure out supporting multiple networks.